api/app/signals/apps/signals/v1/serializers.py

```diff
@@ -383,3 +384,38 @@ def create(self, validated_data):
             priority_data
         )
         return signal
+
+
+class SplitPrivateSignalSerializerList(serializers.ListSerializer):
+    def is_valid(self, raise_exception=False):
+        if (settings.SIGNAL_MIN_NUMBER_OF_CHILDREN > len(self.initial_data) or
+                len(self.initial_data) > settings.SIGNAL_MAX_NUMBER_OF_CHILDREN):
```

Comment:
Bit of a confusing line. I personally like to keep the statements symmetrical for readability, or combine the two statements. Either this:

`(len(self.initial_data) < settings.SIGNAL_MIN_NUMBER_OF_CHILDREN) or
                len(self.initial_data) > settings.SIGNAL_MAX_NUMBER_OF_CHILDREN)`

or

`settings.SIGNAL_MIN_NUMBER_OF_CHILDREN <= len(self.initial_data) <= settings.SIGNAL_MAX_NUMBER_OF_CHILDREN`

The first reads easier, because you would explicitly say that len() is SMALLER than X and GREATER than Y, instead of X larger than len() and len() larger than Y. In the latter case you're comparing two different things, and in your mind you would need to translate this to something you can easily visualise.

The second reads easier because it's immediately clear that len() should be BETWEEN two values.

Maybe just a personal thing, see what you think :)